internal/cli/cli.go

```diff
@@ -46,6 +55,26 @@ func (opts *globalOpts) ProjectID() string {
 	return opts.projectID
 }
 
+type cmdOpt func() error
+
+// PreRunE is a function to call before running the command,
+// this will validate the project ID and call any additional function pass as a callback
+func (opts *globalOpts) PreRunE(cbs ...cmdOpt) error {
```

Comment:
This is our main change, we now have the validation here, and we can pass callbacks with any additional init function that the command may need